Client/Frontend/Reader/ReaderModeCache.swift

```diff
@@ -0,0 +1,84 @@
+        return ReaderModeCacheSharedInstance
+    }
+
+    func put(url: NSURL, _ readabilityResult: ReadabilityResult, error: NSErrorPointer) -> Bool {
+        if let cacheDirectoryPath = cacheDirectoryForURL(url) {
+            if NSFileManager.defaultManager().createDirectoryAtPath(cacheDirectoryPath, withIntermediateDirectories: true, attributes: nil, error: error) {
+                let contentFilePath = cacheDirectoryPath.stringByAppendingPathComponent("content.json")
+                let string: NSString = readabilityResult.encode()
+                return string.writeToFile(contentFilePath, atomically: true, encoding: NSUTF8StringEncoding, error: error)
+            }
+        }
+        return false
+    }
+
+    func get(url: NSURL, error: NSErrorPointer) -> ReadabilityResult? {
+        if let cacheDirectoryPath = cacheDirectoryForURL(url) {
+            let contentFilePath = cacheDirectoryPath.stringByAppendingPathComponent("content.json")
+            if NSFileManager.defaultManager().fileExistsAtPath(contentFilePath) {
+                if let string = NSString(contentsOfFile: contentFilePath, encoding: NSUTF8StringEncoding, error: error) {
+                    return ReadabilityResult(string: string)
+                }
+            }
+        }
+        return nil
+    }
+
+    func delete(url: NSURL, error: NSErrorPointer) {
+        if let cacheDirectoryPath = cacheDirectoryForURL(url) {
+            if NSFileManager.defaultManager().fileExistsAtPath(cacheDirectoryPath) {
+                NSFileManager.defaultManager().removeItemAtPath(cacheDirectoryPath, error: error)
+            }
+        }
+    }
+
+    private func cacheDirectoryForURL(url: NSURL) -> NSString? {
+        if let paths = NSSearchPathForDirectoriesInDomains(NSSearchPathDirectory.CachesDirectory, NSSearchPathDomainMask.UserDomainMask, true) as? [String] {
+            if paths.count > 0 {
+                if let hashedPath = hashedPathForURL(url) {
+                    return paths[0] + "/" + hashedPath
+                }
+            }
+        }
+        return nil
+    }
+
+    private func hashedPathForURL(url: NSURL) -> NSString? {
+        if let hash = hashForURL(url) {
+            return hash.substringWithRange(NSMakeRange(0, 2)) + "/" + hash.substringWithRange(NSMakeRange(2, 2)) + hash.substringFromIndex(4)
+        }
+        return nil
+    }
+
+    private func hashForURL(url: NSURL) -> NSString? {
+        if let absoluteString = url.absoluteString {
+            if let data = absoluteString.dataUsingEncoding(NSUTF8StringEncoding) {
+                var digest = [UInt8](count:Int(CC_SHA1_DIGEST_LENGTH), repeatedValue: 0)
+                CC_SHA1(data.bytes, CC_LONG(data.length), &digest)
+                let output = NSMutableString(capacity: Int(CC_SHA1_DIGEST_LENGTH))
+                for byte in digest {
+                    output.appendFormat("%02x", byte)
+                }
+                return output
```

Comment:
Heh, I just did this same thing for thumbnails. We actually have `HashExtensions` to make this easier, so this entire block can be reduced to:
```
data.sha1.hexEncodedString
```